### PR TITLE
Full flow integration

### PR DESCRIPTION
**Pull request by ksunandp**
### Changes
Updating flow to latest version and extending it's integration.
Updating existing flow implementation to reflect latest version.
Fixing any errors caused by update.
Replacing react proptypes with flow.

#### Trello Ticket / Github Issue Link
https://trello.com/c/NMSTGTw6/122-issue-691-flow-usage-is-inconsistent-across-in-the-application
https://github.com/redbadger/website-honestly/issues/691

### Test
Review needed. Passes Circle Tests.

#### Can this PR be merged immediately after testing (eg. rebase, infrastructure update)?
Yes.

### CODE REVIEW

File: site/components/event-image/index.js [hunk 1]
```diff
@@ -12,8 +18,9 @@ export default function EventImage({ imgPath, imgAlt }) {
     </div>
   );
 }
-
+/*
 EventImage.propTypes = {
   imgPath: React.PropTypes.string.isRequired,
   imgAlt: React.PropTypes.string.isRequired,
 };
+*/
```
Comment (dpiatek) [inline site/components/event-image/index.js]: We can remove these now no?

File: site/components/event-title/index.js [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+// @flow
+import * as React from 'react';
 import styles from '../../pages/events/events-list/style.css';
 import Link from '../link';
 
-const EventTitle = ({ eventLink, eventTitle }) => (
+type EventTitleProps = {
+  eventLink: {
+    to?: string,
+    children?: React.Node,
```
Comment (dpiatek) [inline site/components/event-title/index.js]: It looks like these might be wrong? Does `Link` work correctly if those values are missing?

Comment (dpiatek) [inline site/components/event-title/index.js]: It looks like these might be wrong? Does `Link` work correctly if those values are missing?

File: site/components/header/small-screen-nav.js [hunk 3]
```diff
@@ -14,31 +14,36 @@ const trackAnalytics = title => () =>
...
     document.body.style.position = 'fixed';
     this.smallScreenNav.scrollTop = 0;
   };
 
   documentBodyRelease = () => {
+    // $FlowIgnore
     document.getElementById('mainContent').removeAttribute('aria-hidden');
+    // $FlowIgnore
     document.body.style.position = 'relative';
   };
```
Comment (dpiatek) [inline site/components/header/small-screen-nav.js]: Think we can fix this without ignoring see https://github.com/redbadger/website-honestly/blob/circle-2.0/site/components/header/small-screen-nav.js#L31

File: site/components/header/small-screen-nav.js [hunk 4]
```diff
@@ -81,6 +86,7 @@ export default class SmallScreenNav extends React.Component {
           <div className={styles.smallScreenNavMargin} onClick={this.closeMenu} />
           <div
             ref={c => {
+              // $FlowIgnore
               this.smallScreenNav = c;
```
Comment (dpiatek) [inline site/components/header/small-screen-nav.js]: putting `()` around the assignment should help here

File: site/components/hr/index.js [hunk 5]
```diff
@@ -2,12 +2,12 @@ import React, { Component } from 'react';
 import classNames from 'classnames';
```
Comment (dpiatek) [inline site/components/hr/index.js]: missing `// @flow` here

File: site/pages/home/newsletter-slice/index.js [hunk 6]
```diff
@@ -61,7 +69,7 @@ class NewsLetter extends Component {
     }
   };
 
-  updateUser = data => {
+  updateUser = (data: any) => {
```
Comment (dpiatek) [inline site/pages/home/newsletter-slice/index.js]: Can we do better here then `any`? It's pretty much a soft bail out from the type system.

File: site/components/event-title/index.js (hunk 2 above)
Comment (ksunandp) [reply site/components/event-title/index.js]: I don't know if Link will work with truthfully as no tests associated. My guess would be no, however was no isRequired associated with react prop types. Happy to change though, thoughts?

File: site/components/header/small-screen-nav.js (hunk 3 above)
Comment (ksunandp) [reply site/components/header/small-screen-nav.js]: The problem is there are null checks and that's why it's falling over. Your file referenced does have those checks. I didn't want to edit any implementation and only the typings. Happy to go back and change?

Comment (dpiatek) [reply site/components/header/small-screen-nav.js]: yup lets change it!

Comment (dpiatek) [reply site/components/header/small-screen-nav.js]: Think there is no avoiding that really with the way the typings work. I'd rather fix it now to be honest.

**Comment by ksunandp** [comment]
Requested changes from review done, commited and pushed.

**Pull request merged**